pkg/agent/multicast/mcast_controller.go

```diff
@@ -537,3 +532,70 @@ func getGroupEventKey(obj interface{}) (string, error) {
 	groupState := obj.(*GroupMemberStatus)
 	return groupState.group.String(), nil
 }
+
+func (c *Controller) GetIGMPReportNPStats() map[v1beta2.NetworkPolicyType]map[apitypes.UID]map[string]*types.RuleMetric {
+	return c.igmpSnooper.getStatsMap()
+}
+
+func (c *Controller) GetGroupPods() map[string][]v1beta2.PodReference {
+	groupPodsMap := make(map[string][]v1beta2.PodReference)
+	for _, obj := range c.groupCache.List() {
+		status := obj.(*GroupMemberStatus)
+		members := make([]v1beta2.PodReference, 0)
+		for s := range status.localMembers {
+			iface, found := c.ifaceStore.GetInterfaceByName(s)
+			if found {
+				members = append(members, v1beta2.PodReference{Name: iface.PodName, Namespace: iface.PodNamespace})
+			}
+		}
+		if len(members) > 0 {
+			groupPodsMap[status.group.String()] = members
+		}
+	}
+	return groupPodsMap
+}
+
+// PodTrafficStats encodes the inbound and outbound multicast statistics of each Pod.
+type PodTrafficStats struct {
+	Inbound, Outbound uint64
+}
+
+func (c *Controller) GetPodStats(podName string, podNamespace string) map[*interfacestore.InterfaceConfig]*PodTrafficStats {
```

Comment:
Because `GetContainerInterfacesByPod` get called in this func.

```
// GetContainerInterfacesByPod retrieves InterfaceConfigs for the Pod.
// It's possible that more than one container interface (with different containerIDs) has the same Pod namespace and
// name temporarily when the previous Pod is being deleted and the new Pod is being created almost simultaneously.
// https://github.com/antrea-io/antrea/issues/785#issuecomment-642051884
```

Reply:
This is a temporary state during Pod recreation, I think we don't need to return multiple stats to users which may be confusing. Most consumers of GetContainerInterfacesByPod just use the first interface.

Reply:
updated